### DISCUSSION TITLE
Query: LEFT JOIN not working in BB Query

### DISCUSSION THREAD

**Discussion by Al-Blu** [Budibase/budibase#5273] [Help]
Reference to discussion #5176

I recently got a solution from @melohagan to a QUERY I was struggling with. However it seems there is a 'JOIN' in the query solution where I would need a 'LEFT JOIN'. But the LEFT JOIN doesn't seem to be accepted by BudiBase. Notice in my screenshot that the 'LEFT' doesn't have the keyword color. Is there a way around this??
Thanks

### Solution suggestion was 
_The SQL Query returns use cases that have **not received** any survey responses for the given USER_ID binding:_

_SELECT idUseCases, Name, Description, COUNT(idUseCases) FROM (
SELECT DISTINCT UseCases.*,
CASE WHEN SurveyResults.idUser = '{{ USER_ID }}' THEN 1
END AS UseCaseUserMatch
FROM sys.UseCases
**JOIN** sys.SurveyResults on SurveyResults.idUseCases = UseCases.idUseCases) as alias
GROUP BY idUseCases
HAVING COUNT(idUseCases) < 2_

**Image show the Query result is UseCase._id == 1 but I need UseCase._id 2, 3, 4**

<img width="1182" alt="Screenshot 2022-04-06 at 16 20 59" src="https://user-images.githubusercontent.com/99728863/161999980-5d058a9a-5e11-43e3-a571-556a4098eb44.png">

**Comment by melohagan** [comment]
Hi @Al-Blu 

The syntax colouring isn't always perfect within the budibase UI. For example the keyword **END** is also not highlighted. This is of no functional concern however as the query is sent off to the MySQL database to be handled there. 

I'm sorry the query wasn't what you were looking for. I thought the idea was to only show use cases that the user hadn't completed a survey for. E.g. a task list.

If you are able to create a SQL query in MySQL Workbench that returns the data you want, then it should return the same data within budibase.

**Reply by Al-Blu** [reply]
Hi @melohagan 
_I thought the idea was to only show use cases that the user hadn't completed a survey for. E.g. a task list._

Well that is what i am trying to do. The list of UseCases what still need to be worked through. But the 'JOIN' gives me the UseCase that  the User has completed. Unless I did something wrong, which I don't think so. I will try recreate it MySQL Workbench. 
Then again, 'LEFT JOIN' should have given me 3 results and not 4.........any idea why that would be....
